contracts/src/factory/HyperdriveCreate3Factory.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.22;
+
+import { CREATE3 } from "solmate/utils/CREATE3.sol";
+
+/// @author DELV
+/// @title HyperdriveCreate3Factory
+/// @notice Uses CREATE3 to deploy a contract to a precomputable address that
+///         is only dependent on the deployer's address and the provided salt.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract HyperdriveCreate3Factory {
+    /// @notice Deploy the contract with the provided creation code to a
+    ///         precomputable address determined by only the deployer address
+    ///         and salt.
+    /// @param _salt Salt to use for the deployment.
+    /// @param _creationCode Creation code of the contract to deploy.
+    function deploy(
+        bytes32 _salt,
+        bytes memory _creationCode
+    ) external payable returns (address deployed) {
```

Comment:
To fix the issue that you had with admin controls, let's add a `bytes memory _initializationData` parameter. If this is non-empty, you'll call back into the deployed contract. This will let you update the admin address (for example).